core/src/databases_store/gcs_background.rs

```diff
@@ -0,0 +1,136 @@
+use anyhow::{anyhow, Result};
+use async_std::stream::StreamExt;
+use cloud_storage::{ListRequest, Object};
+use std::collections::HashMap;
+use tracing::error;
+
+use crate::{
+    databases::{
+        csv::GoogleCloudStorageCSVContent,
+        table::{Row, Table},
+        table_schema::TableSchema,
+    },
+    databases_store::gcs::write_rows_to_csv_helper,
+    utils,
+};
+
+// This Store is used to pass upsert and delete call info from the APIs to the background worker
+pub struct GoogleCloudStorageBackgroundProcessingStore {}
+
+impl GoogleCloudStorageBackgroundProcessingStore {
+    fn get_bucket() -> Result<String> {
+        match std::env::var("DUST_UPSERT_QUEUE_BUCKET") {
```

Comment:
Hmmm not sure we want to reuse this one? You're reusing for the retention policy? I worry we might change it without knowing we're impacting this system?

Reply:
Yes, frankly it was a path-of-least-resistance thing, and it's probably better to create a new bucket. I need to learn how to do that, for dev, in prod, ...

> You're reusing for the retention policy?

We do want a retention policy to clean up some old unprocessed junk (should not happen in theory!). What's the retention policy on `DUST_UPSERT_QUEUE_BUCKET`?